test/test_return.py

```diff
@@ -178,6 +178,21 @@ def test_return_location_edit(self):
             == "/#%23foo%3Fbar"
         ), 'location with a variable with encoding'
 
+        assert 'success' in self.conf(
+            '"https://${host}${uri}path$$sub"', 'routes/0/action/location'
+        ), 'location with $ sub-delimiter'
+        assert self.get()['headers']['Location'] == 'https://localhost/path$sub'
+
+        assert 'success' in self.conf(
+            '"https://$host$uri/$$sub1$$sub2"', 'routes/0/action/location'
+        ), 'location with 2 $ sub-delimiters'
+        assert self.get()['headers']['Location'] == 'https://localhost//$sub1$sub2'
+
+        assert 'success' in self.conf(
+            '"https://$host/routes$uri$method$$sub"', 'routes/0/action/location'
```

Comment:
Would you add some test with `$$$somevar` to make sure we don't break that?

Reply:
Sure, will do.